tests/test_on_data/test_format_converters/test_spikeglx_converter.py

```diff
@@ -0,0 +1,69 @@
+import datetime
+from unittest import TestCase
+from shutil import rmtree
+from tempfile import mkdtemp
+from pathlib import Path
+from datetime import datetime
+from copy import deepcopy
+
+from pynwb import NWBHDF5IO
+from pydantic import FilePath
+
+from neuroconv import ConverterPipe
+from neuroconv.converters import SpikeGLXConverter
+from neuroconv.utils import load_dict_from_file
+
+from ..setup_paths import ECEPHY_DATA_PATH
+
+SPIKEGLX_PATH = ECEPHY_DATA_PATH / "spikeglx"
+
+
+class TestSingleProbeSpikeGLXConverter(TestCase):
+    maxDiff = None
+
+    def setUp(self):
+        self.tmpdir = Path(mkdtemp())
+
+    def tearDown(self):
+        rmtree(self.tmpdir)
+
+    def assertNWBFileStructure(self, nwbfile_path: FilePath):
+        with NWBHDF5IO(path=nwbfile_path) as io:
+            nwbfile = io.read()
+
+            expected_session_start_time = datetime(2020, 11, 3, 10, 35, 10).astimezone()
+            assert nwbfile.session_start_time == expected_session_start_time
+
+            assert "ElectricalSeriesAP" in nwbfile.acquisition
+            assert "ElectricalSeriesLF" in nwbfile.acquisition
+            assert "ElectricalSeriesNIDQ" in nwbfile.acquisition
+
+            assert "Neuropixel-Imec" in nwbfile.devices
+
+            assert "NIDQChannelGroup" in nwbfile.electrode_groups
+            assert "s0" in nwbfile.electrode_groups
+
+    def test_single_probe_spikeglx_converter(self):
+        converter = SpikeGLXConverter(folder_path=SPIKEGLX_PATH / "Noise4Sam_g0")
+        # import json
+        metadata = converter.get_metadata()
+
+        test_metadata = deepcopy(metadata)
+        for exclude_field in ["session_start_time", "identifier"]:
+            test_metadata["NWBFile"].pop(exclude_field)
+        expected_metadata = load_dict_from_file(file_path=Path(__file__).parent / "single_probe_metadata.json")
+        self.assertDictEqual(d1=test_metadata, d2=expected_metadata)
+
+        nwbfile_path = self.tmpdir / "test_spikeglx_converter.nwb"
+        converter.run_conversion(nwbfile_path=nwbfile_path, metadata=metadata)
+
+        self.assertNWBFileStructure(nwbfile_path=nwbfile_path)
+
+    def test_in_converter_pipe(self):
+        spikeglx_converter = SpikeGLXConverter(folder_path=SPIKEGLX_PATH / "Noise4Sam_g0")
+        converter_pipe = ConverterPipe(data_interfaces=[spikeglx_converter])
+
+        nwbfile_path = self.tmpdir / "test_spikeglx_converter_in_converter_pipe.nwb"
+        converter_pipe.run_conversion(nwbfile_path=nwbfile_path, overwrite=True)
+
+        self.assertNWBFileStructure(nwbfile_path=nwbfile_path)
```

Comment:
Will this work in NWBConverter?

Reply:
It will not, and after thinking about it it shouldn't be expected to

This is the first time it's come up, but an `NWBConverter `simply cannot, by its design, take a data interface or subconverter that is not designed around the notion of having all information available ahead of time via class methods

That was the point of the `ConverterPipe`, where information is collected and added on a dynamic basis

Specifically, the `get_schema`s and related methods would need to be able to know the `folder_path` that contains the spikeglx sessios, and needs to know how many streams you intend to load, which determines the metadata structure around how many of which streams you might have, etc. But all the `get_schema` class methods are designed to be agnostic to the input arguments of the `__init__`, so this use case doesn't make sense

In general, subconverters like this made from a `ConverterPipe` can only be run from a unifying `ConverterPipe` unless we start changing internal design  a bit more radically

Reply:
**Unless** - we prohibit control over sub-stream specification and the role of the `SpikeGLXConverter` is always by default to load in as many streams as possible - and through the process we keep the conversion options fixed so they cannot be specified for individual streams and only apply globally to all the IMEC/NIDQ 

Then it might be possible